### DISCUSSION TITLE
"Enthusiast" is not a good default setting for `eldoc-documentation-strategy`

### DISCUSSION THREAD

**Discussion by samclearman** [joaotavora/eglot#1032] [Q&A] [answered]
Eglot sets `eldoc-documentation-strategy` to `eldoc-documentation-enthusiast` when it manages a buffer.  In typescript, at least, this means that when you put the point over an error, you can't see what the error is.  I think `eldoc-documentation-compose-eagerly` would be better?  And/or make this configurable.

**Comment by joaotavora** [comment, marked as answer]
It _is_ configurable.  Just set it in `eglot-managed-mode-hook`. And this has been discussed often in this bug tracker (like https://github.com/joaotavora/eglot/issues/648) If you need to fix an error in the function call, it's better to see the arguments than to see the error, and that's why it is the default (if there were more lines available, a different default could have been chosen). You can see the error in other ways, see #898.

Also, if I'm not mistaken, you completely ignored some very capitalized instructions about _not_ opening an issue for this.  Why?

**Reply by samclearman** [reply]
Thanks so much for such a helpful reply!

I didn't see the instructions that you're talking about.  I assume you're talking about the ones in your issue template, which I just discovered.  The reason that I didn't see those is that I got to issues/new by clicking "reference in new issue" on the line of code which sets the value.  Apparently when you do this it doesn't show you the issue template, which I didn't know until now.

Anyway, sorry about that and thanks again.

**Reply by joaotavora** [reply]
> The reason that I didn't see those is that I got to issues/new by clicking "reference in new issue" on the line of code which sets the value. Apparently when you do this it doesn't show you the issue template, which I didn't know until now.

That's good to know, didn't know about that.  I thought it was impossible to skip the instructions.  Maybe I should stop berating people :-)

**Reply by samclearman** [reply]
No worries :)
